ssl/rio/rio_notifier.c

```diff
@@ -15,14 +17,45 @@ static int set_cloexec(int fd)
 #endif
 }
 
+# if defined(OPENSSL_SYS_WINDOWS)
+
+static CRYPTO_ONCE ensure_wsa_startup_once = CRYPTO_ONCE_STATIC_INIT;
+static int wsa_started;
+
+static void ossl_wsa_cleanup(void)
+{
+    if (wsa_started) {
+        wsa_started = 0;
+        WSACleanup();
+    }
+}
+
+DEFINE_RUN_ONCE_STATIC(do_wsa_startup)
+{
+    WORD versionreq = 0x0202; /* Version 2.2 */
+    WSADATA wsadata;
+
+    if (WSAStartup(versionreq, &wsadata) != 0)
+        return 0;
+    wsa_started = 1;
+    OPENSSL_atexit(ossl_wsa_cleanup);
+    return 1;
+}
+
+static int ensure_wsa_startup(void)
```

Comment:
Not sure why this needs to be a separate function. It seems overkill. But if we keep it lets at least inline it.

Reply:
Made it inline.